src/__tests__/utils/url_utils.test.js

```diff
@@ -0,0 +1,58 @@
+import { getOriginFromUrl, getLocationFromUrl } from '../../utils/url_utils';
+
+describe('url utils', () => {
+  describe('getOriginFromUrl', function() {
+    it('should return undefined if there is no url', function() {
+      expect(getOriginFromUrl()).toBe(undefined);
+      expect(getOriginFromUrl('')).toBe(undefined);
+      expect(getOriginFromUrl(null)).toBe(undefined);
+    });
+    it('should use an anchor to parse the url and return the origin', function() {
+      var url = 'https://test.com/example';
+      expect(getOriginFromUrl(url)).toBe('https://test.com');
+    });
+    it('should use add the `port` when available', function() {
+      var url = 'https://localhost:3000/example';
+      expect(getOriginFromUrl(url)).toBe('https://localhost:3000');
+    });
+  });
+  describe('getLocationFromUrl', function() {
+    const mapping = {
+      'https://localhost:3000/foo?id=1': {
+        href: 'https://localhost:3000/foo?id=1',
+        protocol: 'https:',
+        host: 'localhost:3000',
+        hostname: 'localhost',
+        port: '3000',
+        pathname: '/foo',
+        search: '?id=1',
+        hash: ''
+      },
+      'https://auth0.com/foo': {
+        href: 'https://auth0.com/foo',
+        protocol: 'https:',
+        host: 'auth0.com',
+        hostname: 'auth0.com',
+        port: undefined,
+        pathname: '/foo',
+        search: '',
+        hash: ''
+      },
+      'https://auth0.com#access_token=foo': {
```

Comment:
If I enter manage.auth0.com, the `#` is put after the `/`.  .e.g.: `https://manage.auth0.com/#/applications`

Reply:
/# and # is the same thing

Reply:
I tried:
- https://manage.auth0.com/#/applications -> fine
- https://manage.auth0.com#/applications -> redirs to https://manage.auth0.com/#/applications
- https://manage.auth0.com/#applications -> no content in the frame
- https://manage.auth0.com#applications -> no content in the frame

So as how I see it, I don't think they are the same. Maybe I'm wrong? Anyway, just add an extra case to test that scenario where the `#` is after the `/` such as `https://manage.auth0.com/#/applications`.